src/widgets/survey/ui/Finish.tsx

```diff
@@ -41,21 +41,30 @@ function FinishItem({
       appletId,
       activityId,
       eventId,
+      order,
     });
 
+  const { isTimerElapsed } = useFlowState({
+    appletId,
+    eventId,
+    flowId,
+  });
+
   const {
     mutate: sendAnswers,
     isLoading: isSendingAnswers,
     isSuccess: successfullySentAnswers,
     isPaused: isOffline,
   } = useActivityAnswersMutation({
     onError: error => {
-      if (error.evaluatedMessage) {
+      if (error.response.status !== 401 && error.evaluatedMessage) {
```

Comment:
Will it be redirected to login screen exactly at this moment ?

Reply:
Automatic redirection to the Logic screen is handled on `RootNavigator.tsx`
<img width="468" alt="image" src="https://user-images.githubusercontent.com/20073193/234845727-938970af-3a8b-4e01-bb0d-5bd800850301.png">